python/vyos/ifconfig/interface.py

```diff
@@ -951,6 +951,9 @@ def set_dhcp(self, enable):
         pid_file = f'{config_base}_{ifname}.pid'
         lease_file = f'{config_base}_{ifname}.leases'
 
+        # Stop client with old config files to get the right IF_METRIC.
+        self._cmd(f'systemctl stop dhclient@{ifname}.service')
```

Comment:
It does not look safe because it stops even before the first start and when `DHCLIENT_OPTS` still defaults.

Reply:
It does not execute the stop command if the service is not started. Tested with:

```
sudo systemctl stop dhclient@eth0.service
sudo tee /etc/systemd/system/dhclient@.service <<'EOF' >/dev/null
[Unit]
Description=DHCP client on %i
Documentation=man:dhclient(8)
ConditionPathExists=/var/lib/dhcp/dhclient_%i.conf
ConditionPathExists=/var/lib/dhcp/dhclient_%i.options
After=vyos-router.service

[Service]
WorkingDirectory=/var/lib/dhcp
Type=exec
EnvironmentFile=-/var/lib/dhcp/dhclient_%i.options
PIDFile=/var/lib/dhcp/dhclient_%i.pid
ExecStart=/sbin/dhclient -4 $DHCLIENT_OPTS
ExecStop=/bin/logger "Testing: stopping DHCP client on %i"
ExecStop=/sbin/dhclient -4 $DHCLIENT_OPTS -r
Restart=always

[Install]
WantedBy=multi-user.target
EOF
sudo systemctl daemon-reload
sudo systemctl stop dhclient@eth0.service
grep "Testing:" /var/log/messages
sudo systemctl start dhclient@eth0.service
sudo systemctl stop dhclient@eth0.service
grep "Testing:" /var/log/messages
```